docs/applications/consuming-android-libraries.md

```diff
@@ -0,0 +1,196 @@
+---
+id: consuming-android-libraries
+title: Consuming Android libraries
+sidebar_label: Consuming Android libraries
+---
+
+# Consuming Android libraries
+
+## Megazord (composite) libraries
+
+The Rust component libraries that Application Services publishes stand alone: each published Android
+ARchive (AAR) contains managed code (`classes.jar`) and multiple `.so` library files (one for each
+supported architecture).  That means consuming multiple such libraries entails at least two `.so`
+libraries, and each of those libraries includes the entire Rust standard library as well as
+(potentially many) duplicated dependencies.  To save space and allow cross-component native-code
+Link Time Optimization (LTO, i.e., inlining, dead code elimination, etc) Application Services also
+publishes composite libraries -- so called *megazord libraries* -- that compose multiple Rust
+components into a single optimized `.so` library file.  The managed code can be easily configured to
+use such a megazord without significant changes.
+
+There are two tasks we want to arrange.  First, we want to substitute component modules for the single
+megazord module; second, we want to arrange for native Rust code to be available to JVM unit tests.
+
+Both tasks are handled by the [megazord-gradle](https://github.com/mozilla/megazord-gradle) Gradle
+plugin; see that page for details on how to apply the plugin to your build configuration.
+
+## Consuming megazords
+
+To consume a specific megazord module, use something like:
+
+```groovy
+appservices {
+    defaultConfig {
+        // Megazord in all variants.  The default is to not megazord.
```

Comment:
This line uses "megazord" as a verb, which is a change from how its used in the introductory text.  It may be worth adding a brief note or explanation of this usage in the introduction, along the lines of "We call this process 'megazording'".